python_modules/dagster/dagster_tests/storage_tests/test_asset_status_cache.py

```diff
@@ -316,3 +316,44 @@ def asset1():
         )
         assert cached_status
         assert cached_status.serialized_materialized_partition_subset is None
+
+
+def test_failure_cache():
+    partitions_def = StaticPartitionsDefinition(["good1", "good2", "fail1", "fail2"])
+
+    @asset(partitions_def=partitions_def)
+    def asset1(context):
+        if context.partition_key.startswith("fail"):
+            raise Exception()
+
+    asset_key = AssetKey("asset1")
+    asset_graph = AssetGraph.from_assets([asset1])
+    asset_job = define_asset_job("asset_job").resolve([asset1], [])
+
+    with instance_for_test() as created_instance:
```

Comment:
can we also test what the cache value looks like when no events exist?

I think it's also worth adding a test for the unpartitioned case, to assert that both the failed subset and materialized subset are `None`. Those values should only be `empty_subset`s when unpartitioned, or partitioned but never updated.